src/rpc/fts/analyzer/FourthOrderParameter.tpp

```diff
@@ -126,6 +126,11 @@ namespace Rpc {
    template <int D>
    void FourthOrderParameter<D>::computeFourthOrderParameter()
    {
+      if (!simulator().hasWc()){
+         system().compute();
```

Comment:
I don't think you should call system().compute() unless: (a) You need the solution to the modified diffusion equation, which yield s the cFields and (b) you know that the solution to the modified diffusion equation is not up to date. The second question can be tested by calling system().hasCFields(). I think this and analogous code in other analyzers that need the cFields, and thus need an up to date solution of the modified diffusion equation, should be

if (!system().hasCFields()) {
     system().compute();
}
if (!simulator().hasWc()) {
    simulator().computeWc();
}

It's not clear to me, however, that the FourthOrderParameter analyzer actually needs the c fields - it appears that the order parameter is entirely based on analysis of the w field components in an eigenvector basis.  Could you just skip the call to system().compute() entirely in this analyzer?

If an analyzer needs other quantities, such as free energy, stress, Cc components, Dc components, etc. you might need to add more tests and compute functions as needed.  

The need to do such a extensive tests is a bit of a pain, but probably unavoidable for analyzers that are designed to work in postprocessing, since we don't do any calculations after reading in each trajectory frame by default.

Please add analogous code to rpg analyzers.